src/main/java/com/amazonaws/services/sqs/AmazonSQSIdleQueueDeletingClient.java

```diff
@@ -114,20 +117,41 @@ protected synchronized void startSweeper(AmazonSQSRequester requester, AmazonSQS
         if (this.idleQueueSweeper != null) {
             throw new IllegalStateException("Idle queue sweeper is already started!");
         }
-        
+
+        // Create the DLQ first so the primary queue can reference it
+        deadLetterQueueUrl = createOrUpdateQueue(queueNamePrefix + SWEEPING_QUEUE_DLQ_SUFFIX, Collections.emptyMap());
```

Comment:
Should DLQ use same SSE settings? Also, it would be nice to have a longer retention period on DLQ.

Reply:
Technically SSE on the DLQ is irrelevant since no messages will ever be sent there directly, and the messages moved there from the main queue will already be encrypted.

Agreed about the retention period though, I'll make it the maximum 14 days.